src/Editor/GUI/Editors/CurveEditor.cpp

```diff
@@ -0,0 +1,749 @@
+            default:
+            case TYPE::LINEAR: {
+                return p;
+            }
+
+            // Modeled after the parabola y = x^2
+            case TYPE::QUADIN: {
+                return p * p;
+            }
+
+            // Modeled after the parabola y = -x^2 + 2x
+            case TYPE::QUADOUT: {
+                return -(p * (p - 2));
+            }
+
+            // Modeled after the piecewise quadratic
+            // y = (1/2)((2x)^2)             ; [0, 0.5)
+            // y = -(1/2)((2x-1)*(2x-3) - 1) ; [0.5, 1]
+            case TYPE::QUADINOUT: {
+                if (p < 0.5)
+                    return 2 * p * p;
+                else
+                    return (-2 * p * p) + (4 * p) - 1;
+            }
+
+            // Modeled after the cubic y = x^3
+            case TYPE::CUBICIN: {
+                return p * p * p;
+            }
+
+            // Modeled after the cubic y = (x - 1)^3 + 1
+            case TYPE::CUBICOUT: {
+                double f = (p - 1);
+                return f * f * f + 1;
+            }
+
+            // Modeled after the piecewise cubic
+            // y = (1/2)((2x)^3)       ; [0, 0.5)
+            // y = (1/2)((2x-2)^3 + 2) ; [0.5, 1]
+            case TYPE::CUBICINOUT: {
+                if (p < 0.5)
+                    return 4 * p * p * p;
+                else {
+                    double f = ((2 * p) - 2);
+                    return 0.5 * f * f * f + 1;
+                }
+            }
+
+            // Modeled after the quartic x^4
+            case TYPE::QUARTIN: {
+                return p * p * p * p;
+            }
+
+            // Modeled after the quartic y = 1 - (x - 1)^4
+            case TYPE::QUARTOUT: {
+                double f = (p - 1);
+                return f * f * f * (1 - p) + 1;
+            }
+
+            // Modeled after the piecewise quartic
+            // y = (1/2)((2x)^4)        ; [0, 0.5)
+            // y = -(1/2)((2x-2)^4 - 2) ; [0.5, 1]
+            case TYPE::QUARTINOUT: {
+                if (p < 0.5)
+                    return 8 * p * p * p * p;
+                else {
+                    double f = (p - 1);
+                    return -8 * f * f * f * f + 1;
+                }
+            }
+
+            // Modeled after the quintic y = x^5
+            case TYPE::QUINTIN: {
+                return p * p * p * p * p;
+            }
+
+            // Modeled after the quintic y = (x - 1)^5 + 1
+            case TYPE::QUINTOUT: {
+                double f = (p - 1);
+                return f * f * f * f * f + 1;
+            }
+
+            // Modeled after the piecewise quintic
+            // y = (1/2)((2x)^5)       ; [0, 0.5)
+            // y = (1/2)((2x-2)^5 + 2) ; [0.5, 1]
+            case TYPE::QUINTINOUT: {
+                if (p < 0.5)
+                    return 16 * p * p * p * p * p;
+                else {
+                    double f = ((2 * p) - 2);
+                    return  0.5 * f * f * f * f * f + 1;
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') keyword 'return' not immediately followed by a semicolon or a single space [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT006)